w3af/plugins/auth/autocomplete_js.py

```diff
@@ -256,22 +273,71 @@ def _find_form_submit_strategy(self, chrome, form):
 
         return None
 
-    def has_active_session(self, debugging_id=None):
+    def has_active_session(self, debugging_id=None, chrome=None):
         """
         Check user session with chrome
         """
         has_active_session = False
         self._set_debugging_id(debugging_id)
-        chrome = self._get_chrome_instance(load_url=False)
+        if not chrome:
+            chrome = self._get_chrome_instance(load_url=False)
 
         try:
             chrome.load_url(self.check_url)
-            chrome.wait_for_load()
+            loaded = chrome.wait_for_load()
+            if not loaded:
+                msg = 'Failed to load %s in chrome for autocomplete_js'
+                args = (self.check_url,)
+                self._log_debug(msg % args)
+                return False
             has_active_session = self.check_string in chrome.get_dom()
         finally:
             chrome.terminate()
             return has_active_session
 
+    def get_options(self):
+        """
+        :returns OptionList: list of option objects for plugin
+        """
+        option_list = super(autocomplete_js, self).get_options()
+        autocomplete_js_options = [
+            (
+                'username_field_css_selector',
+                self.username_field_css_selector,
+                STRING,
+                "(Optional) Exact CSS selector which will be used to retrieve "
+                "the username input field. If provided then w3af won't try "
+                "to detect input field automatically."
+            ),
+            (
+                'login_button_css_selector',
+                self.login_button_css_selector,
+                STRING,
+                "(Optional) Exact CSS selector which will be used to retrieve "
+                "the login button field. If provided then w3af won't try "
+                "to detect button automatically."
+            ),
+        ]
+        for option in autocomplete_js_options:
+            option_list.add(opt_factory(
+                option[0],
+                option[1],
+                option[3],
+                option[2],
+                help=option[3],
+            ))
+        return option_list
+
+    def set_options(self, options_list):
+        options_list_copy = deepcopy(options_list)  # we don't want to touch real option_list
+        self.username_field_css_selector = options_list_copy.pop(
+            'username_field_css_selector'
+        ).get_value()
+        self.login_button_css_selector = options_list_copy.pop(
+            'login_button_css_selector'
+        ).get_value()
+        super(autocomplete_js, self).set_options(options_list_copy)
+
```

Comment:
Add the new configuration parameters to `get_long_desc`

Add some documentation explaining how they work, when to use them, etc.